sorting/binary_insertion_sort.c

```diff
@@ -3,6 +3,9 @@
  * inserting the selected item at each iteration. */
 #include <stdio.h>
 #include <stdlib.h>
+#include <assert.h>
+
+#define ARR_SZ_CAP	1024
```

Comment:
Could you explain the use of this variable, please?

Reply:
The size of the input arrays is taken by the user and memory is directly allocated depending on that size. This is usually not safe as the user might set a very large value and without any sanitation of input, the program would try to allocate a very large memory and might have unexpected behavior. 

Assert makes sure that the input size is less than the max capacity, otherwise safely exits the program,

Reply:
**This is commit where I added it and the reason**

[Safely allocation a fixed max size of array as opposed to directly taking size as input which may be unsafe](https://github.com/TheAlgorithms/C/pull/1031/commits/a97cd20a8240eaf8bbb84dfc6416f95367377871) 
a97cd20